x/twap/logic_test.go

```diff
@@ -521,3 +522,422 @@ func (s *TestSuite) TestPruneRecords() {
 
 	s.validateExpectedRecords(expectedKeptRecords)
 }
+
+// TestUpdateRecords tests that the records are updated correctly.
+// It tests the following:
+// - two-asset pools
+// - multi-asset pools
+// - with spot price errors
+// - without spot price errors
+// - that new records are created
+// - older historical records are not updated
+// - spot price errot times are either propagated from
+// older records or set to current block time in case error occurred.
+func (s *TestSuite) TestUpdateRecords() {
+
+	type spOverride struct {
+		poolId      uint64
+		baseDenom   string
+		quoteDenom  string
+		overrideSp  sdk.Dec
+		overrideErr error
+	}
+
+	type expectedResults struct {
+		spotPriceA    sdk.Dec
+		spotPriceB    sdk.Dec
+		lastErrorTime time.Time
+	}
+
+	var spError = errors.New("spot price error")
+
+	validateRecords := func(expectedRecords []expectedResults, actualRecords []types.TwapRecord) {
```

Comment:
nit: I think this could be a suite method + can be used in different places around our test code

Reply:
I've given our tests a quick skim and couldn't find where this function and `expectedResults` type could be reused.

The problem here is that `expectedResults` is a type that is local to this test. I think if anyone decides to make that type global / apply it in other tests, then it would make sense.